packages/@webex/plugin-meetings/src/hashTree/hashTree.ts

```diff
@@ -0,0 +1,459 @@
+
+    let offset = 0;
+
+    // iterate through the item types lexicographically
+    const itemTypes = Object.keys(leafContent).sort();
+    itemTypes.forEach((type) => {
+      // iterate through the items of this type in ascending order of ID
+      const items = Object.values(leafContent[type]).sort(
+        (a: LeafDataItem, b: LeafDataItem) => a.id - b.id
+      );
+
+      // add all the items id and version to the hasher
+      items.forEach((item: LeafDataItem) => {
+        buffer.writeBigInt64LE(BigInt(item.id), offset);
+        buffer.writeBigInt64LE(BigInt(item.version), offset + 8);
+
+        offset += 16;
+      });
+    });
+
+    this.leafHashes[index] = XXH3_128(buffer, BigInt(0)).toString(16).padStart(32, '0');
+  }
+
+  /**
+   * Computes all internal and leaf node hashes of the tree.
+   * Internal node hashes are computed bottom-up from the leaf hashes.
+   * @returns {string[]} An array of hash strings, with internal node hashes first, followed by leaf hashes.
+   * Returns `[EMPTY_HASH]` if the tree has 0 leaves.
+   */
+  computeTreeHashes(): string[] {
+    if (this.numLeaves === 0) {
+      return [EMPTY_HASH];
+    }
+
+    let currentLevelHashes = [...this.leafHashes];
+    const allHashes = [];
+
+    while (currentLevelHashes.length > 1) {
+      const nextLevelHashes: string[] = [];
+      for (let i = 0; i < currentLevelHashes.length; i += 2) {
+        const leftHash = currentLevelHashes[i];
+        const rightHash = i + 1 < currentLevelHashes.length ? currentLevelHashes[i + 1] : leftHash;
+
+        const input = Buffer.concat([
+          Buffer.from(leftHash, 'hex').subarray(0, 8).reverse(),
+          Buffer.from(leftHash, 'hex').subarray(8, 16).reverse(),
+          Buffer.from(rightHash, 'hex').subarray(0, 8).reverse(),
+          Buffer.from(rightHash, 'hex').subarray(8, 16).reverse(),
+        ]);
+
+        nextLevelHashes.push(XXH3_128(input, BigInt(0)).toString(16).padStart(32, '0'));
+      }
+      currentLevelHashes = nextLevelHashes;
+      allHashes.unshift(...currentLevelHashes);
+    }
+
+    return [...allHashes, ...this.leafHashes];
+  }
+
+  /**
+   * Returns all hashes in the tree (internal nodes then leaf nodes).
+   * @returns {string[]} An array of hash strings.
+   */
+  getHashes(): string[] {
+    return this.computeTreeHashes();
+  }
+
+  /**
+   * Computes and returns the hash value of the root node.
+   * @returns {string} The root hash of the entire tree. Returns `EMPTY_HASH` if the tree has 0 leaves.
+   */
+  getRootHash(): string {
+    if (this.numLeaves === 0) {
+      return EMPTY_HASH;
+    }
+
+    return this.computeTreeHashes()[0];
+  }
+
+  /**
+   * Gets the number of leaves in the tree.
+   * @returns {number} The number of leaves.
+   */
+  getLeafCount(): number {
+    return this.numLeaves;
+  }
+
+  /**
+   * Calculates the total number of items stored in the tree.
+   * @returns {number} The total number of items.
+   */
+  getTotalItemCount(): number {
```

Comment:
I know this is used in the unit tests quite a bit, but do you think it be worth adding some explicit/"dumb" ```getTotalItemCount``` unit tests?